src/Microsoft.Graph.Core/Requests/AuthenticationHandler.cs

```diff
@@ -103,18 +95,27 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             if (AuthenticationProvider != null)
             {
                 await AuthenticationProvider.AuthenticateRequestAsync(httpRequestMessage);
-            }
 
-            HttpResponseMessage response = await base.SendAsync(httpRequestMessage, cancellationToken);
+                HttpResponseMessage response = await base.SendAsync(httpRequestMessage, cancellationToken);
+
+                // Chcek if response is a 401 & is not a streamed body (is buffered)
+                if (IsUnauthorized(response) && httpRequestMessage.IsBuffered())
+                {
+                    // re-issue the request to get a new access token
+                    response = await SendRetryAsync(response, cancellationToken);
```

Comment:
Do we have innerHandler business rules? Is it possible to have an innerRetry handler? If so, we should use the retryHandler if we have one.

Reply:
Sadly, I don't think MSAL exposes the HttpClient it uses internally, so we can't add handlers to their middleware pipeline.  So we have to do the retry using this kind of wrapper code.